### DISCUSSION TITLE
MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version

### DISCUSSION THREAD

**Discussion by Jeoffreybauvin** [PyMySQL/mysqlclient#651] [Issue report]
### Describe the bug

I have a lot of these : 

`MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version`

Can you fix this deprecated warning ? 

https://dev.mysql.com/doc/c-api/8.0/en/c-api-auto-reconnect.html

### Environment

Mysql 8.0.33
Debian 11

### How did you install libmysqlclient libraries?

Debian package

### What version of mysqlclient do you use?

2.2.0

### Minimum but complete code to reproduce

```python
# Write Python code here.
import MySQLdb

conn = MySQLdb.connect(host='127.0.0.1', port=3306, user='root')
...
```

### Schema and initial data required to reproduce.

```sql
-- Write SQL here.
-- e.g. CREATE TABLE ...
```

**Comment by methane** [comment]
I can not reproduce the warning with your sample code.

**Reply by methane** [reply]
I can reproduce the warning only when using `conn.ping(False)` or `conn.ping(True)`.
You can avoid warning by `conn.ping()`.

**Reply by zzzeek** [reply]
is conn.ping() with no arguments guaranteed to not attempt to reconnect?   The function seems to be incompatible with that of pymysql now, can you clarify ?

**Reply by methane** [reply]
> is conn.ping() with no arguments guaranteed to not attempt to reconnect?

Yes.

https://github.com/PyMySQL/mysqlclient/blob/91c04282eac46ce2e5380be58e262d817faab566/src/MySQLdb/_mysql.c#L1878C1-L1884C6

> The function seems to be incompatible with that of pymysql now, can you clarify ?

Yes, they are incompatible. I didn't notice it.
I will update them:

* Both: Deprecate `reconnect` argument.
* PyMySQL: change the default to false.
* mysqlclient: change to not use MYSQL_OPT_RECONNECT when `reconnect=False` is passed.

But I can not releate them soon.
